### ISSUE TITLE
Loss function for transformers

### ISSUE THREAD

**Issue by saurav935** [adap/flower#1749]
In [this](https://github.com/adap/flower/blob/main/examples/quickstart_huggingface/client.py) example for transformers, what loss function is used for training distilbert?

Is the loss function that is used for performing training at each client and the loss function that FedAvg is trying to minimize different?

**Comment by adam-narozniak** [comment]
Hi!
1. What loss function is used for training DistilBERT?
The loss for DistilBERT is, according to the hugging face documentation: "[...]a triple loss combining language modeling, distillation and cosine-distance losses." ([source](https://huggingface.co/docs/transformers/model_doc/distilbert)).

2. Is the loss function that is used for performing training at each client and the loss function that FedAvg is trying to minimize different?
Yes, it is trying to minimize the same loss, but let me clarify that even further. The minimization function of the FedAvg can be formulated as a weighted sum of local losses. 
But note that in federated learning, no optimization steps with gradient calculation happen on the server side.

I hope this clarifies your questions.

**Comment by saurav935** [comment]
Hi, thanks for your response!
So how does FedAvg minimize the loss on the server side?

**Comment by mfcitycapital** [comment]
Server doesn't minimize the loss function, it just averages the weights of the local models.

**Comment by WilliamLindskog** [comment]
Closing this issue as it seems to have been resolved.
